### ISSUE TITLE
librsvg: DoS in librsvg parsing SVGs with circular definitions

### ISSUE THREAD

**Issue by MingcongBai** [AOSC-Dev/aosc-os-abbs#242]
Original oss-security message:

```
Hello,

Two DoS in librsvg 2.40.2 parsing SVGs with circular definitions were found (they will produce stack exhaustion). Other versions can be vulnerable too. They affect the following functions:

* rsvg_cairo_pop_discrete_layer - rsvg_cairo_pop_render_stack - rsvg_cairo_generate_mask: reproducible using circular-1.svg
* _rsvg_css_normalize_font_size: reproducible using circular-2.svg

Both reproducers are attached in a tar.gz to avoid a crash in my own browser.  Fortunately, these issues are solved in the last git revision of librsvg2.

Regards,
Gustavo.
```

Fixed CVE: CVE-2016-4347, CVE-2016-4348.

Git revision fixes: a51919f7e1ca9c535390a746fbf6e28c8402dc61, d1c9191949747f6dcfd207831d15dd4ba00e31f2. At https://github.com/GNOME/librsvg.

**Comment by MingcongBai** [comment]
Hmm, this is a bigger backport project than I thought...

Given the lower security risk carried by these two CVEs, I would wait for the next tag by the upstream.

**Comment by MingcongBai** [comment]
This should be fixed as per 2.4.16. Announcement: https://mail.gnome.org/archives/ftp-release-list/2016-June/msg00007.html.

Not mentioned in the ChangeLog, but the fix commits are included as the update.

**Comment by MingcongBai** [comment]
Fixed a typo.
